libraries/ts-command-line/src/providers/ScopedCommandLineAction.ts

```diff
@@ -58,6 +59,10 @@ class InternalScopedCommandLineParser extends CommandLineParser {
     this._internalOptions.onDefineScopedParameters(this);
   }
 
+  public _registerDefinedParameters(): void {
+    super._registerDefinedParameters(this._internalOptions.existingParameterNames);
```

Comment:
Passing this as an argument everywhere feels awkward.

Reply:
Its the only way to have access to the parameters that have already been defined from the layer above it, unless you have some other ideas here? Since registration is deferred to the very last minute when `_registerDefinedParameters` is called before calling `parseArgs`, it's really the only opportunity I could find to funnel that information around.

Reply:
We could instead pass around a state object? Though it would likely contain the same data, but would be more "generalized"